### PR TITLE
[many ports] Fix cmake warnings

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?
  Fixes cmake warnings like `PREFER_NINJA` with `vcpkg_cmake_configure` and alike

This PR enables us to turn warnings on `PREFER_NINJA` with `vcpkg_cmake_configure` and warnings on unparsed arguments into errors. It would help to catch common errors like missing `#` after commit id in `vcpkg_from_github` or calling `vcpkg_cmake_configure` with `PREFER_NINJA`.

### CODE REVIEW

**Comment by Thomas1664** [comment]
@BillyONeal Can you please take a look at this vcpkg crash?

```
Failure to remove_all_inside("D:\buildtrees") due to file "\\?\D:\buildtrees\ffmpeg": The process cannot access the file because it is being used by another process.
##[error]vcpkg clean failed
At C:\a\1\s\scripts\azure-pipelines\test-modified-ports.ps1:132 char:5
+     throw "vcpkg clean failed"

```

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/crfsuite/vcpkg.json [hunk 1]
```diff
@@ -1,11 +1,15 @@
 {
   "name": "crfsuite",
-  "version-date": "2019-07-21",
-  "port-version": 3,
+  "version-date": "2020-08-27",
   "description": "CRFSuite is an implementation of Conditional Random Fields (CRFs) for labeling sequential data.",
   "homepage": "https://www.chokkan.org/software/crfsuite/",
+  "license": null,
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/crfsuite/vcpkg.json:6]: Read the full story:
https://github.com/chokkan/crfsuite/blob/master/README

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fixes! Sorry for the CI runaround.

**Pull request merged**